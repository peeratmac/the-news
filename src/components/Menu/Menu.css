nav {
  padding-right: 30px;
  padding-left: 30px;
}

h1 {
  color: #30336b;
}

.the-news-logo {
  font-size: 50px;
  padding-bottom: 30px;
  background: #f6e58d;
  border-radius: 15px;
}

span {
  font-size: 80px;
  color: #ff7979;
}

h2 {
  cursor: pointer;
}

h3 {
  cursor: pointer;
  color: #30336b;
  padding-bottom: 15px;
}

.icon-image {
  border-radius: 0px;
  height: 50px;
  width: 50px;
}
