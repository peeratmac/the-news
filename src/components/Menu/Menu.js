import React from 'react';
import LocalImage from '../../icons/local.svg';
import TechnologyImage from '../../icons/technology.svg';
import EntertainmentImage from '../../icons/entertainment.svg';
import ScienceImage from '../../icons/science.svg';
import HealthImage from '../../icons/health.svg';
import './Menu.css';
import PropTypes from 'prop-types';

const Menu = ({ changeNewsCategory, newsState }) => {
  const { local, technology, entertainment, science, health } = newsState;
  return (
    <nav>
      <h1 className='the-news-logo'>
        the <span>news</span>
      </h1>
      <div className='menu-side-bar'>
        <h2 onClick={() => changeNewsCategory(local)}>
          <img className='icon-image' src={LocalImage} alt='icon' />{' '}
          <p className='category-label'>Local News</p>
        </h2>
        <h2 onClick={() => changeNewsCategory(technology)}>
          <img className='icon-image' src={TechnologyImage} alt='icon' />
          <p className='category-label'>Technology</p>
        </h2>
        <h2 onClick={() => changeNewsCategory(entertainment)}>
          <img className='icon-image' src={EntertainmentImage} alt='icon' />
          <p className='category-label'>Entertainment</p>
        </h2>
        <h2 onClick={() => changeNewsCategory(science)}>
          <img className='icon-image' src={ScienceImage} alt='icon' />
          <p className='category-label'>Science</p>
        </h2>
        <h2 onClick={() => changeNewsCategory(health)}>
          <img className='icon-image' src={HealthImage} alt='icon' />
          <p className='category-label'>Health</p>
        </h2>
      </div>
    </nav>
  );
};

export default Menu;

Menu.propTypes = {
  changeNewsCategory: PropTypes.func.isRequired,
  newsState: PropTypes.object.isRequired
};
