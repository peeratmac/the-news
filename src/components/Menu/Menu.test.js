import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
  it('should match the Menu Snapshot', () => {
    const mockChangeNewsCategory = jest.fn();
    const mockState = {
      selectedNews: [],
      local: [],
      technology: [],
      entertainment: [],
      science: [],
      health: [],
      searchNews: []
    };
    const wrapper = shallow(
      <Menu changeNewsCategory={mockChangeNewsCategory} newsState={mockState} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
