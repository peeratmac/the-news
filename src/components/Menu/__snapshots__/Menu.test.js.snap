// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the Menu Snapshot 1`] = `
<nav>
  <h1
    className="the-news-logo"
  >
    the 
    <span>
      news
    </span>
  </h1>
  <div
    className="menu-side-bar"
  >
    <h2
      onClick={[Function]}
    >
      <img
        className="icon-image"
        src="local.svg"
      />
       
      <h3>
        Local News
      </h3>
    </h2>
    <h2
      onClick={[Function]}
    >
      <img
        className="icon-image"
        src="technology.svg"
      />
      <h3>
        Technology
      </h3>
    </h2>
    <h2
      onClick={[Function]}
    >
      <img
        className="icon-image"
        src="entertainment.svg"
      />
      <h3>
        Entertainment
      </h3>
    </h2>
    <h2
      onClick={[Function]}
    >
      <img
        className="icon-image"
        src="science.svg"
      />
      <h3>
        Science
      </h3>
    </h2>
    <h2
      onClick={[Function]}
    >
      <img
        className="icon-image"
        src="health.svg"
      />
      <h3>
        Health
      </h3>
    </h2>
  </div>
</nav>
`;
